src/transformers/trainer_seq2seq.py

```diff
@@ -271,15 +271,16 @@ def prediction_step(
             gen_kwargs["synced_gpus"] if gen_kwargs.get("synced_gpus") is not None else default_synced_gpus
         )
 
+        generation_inputs = inputs.copy()
         # If the `decoder_input_ids` was created from `labels`, evict the former, so that the model can freely generate
         # (otherwise, it would continue generating from the padded `decoder_input_ids`)
         if (
-            "labels" in inputs
-            and "decoder_input_ids" in inputs
-            and inputs["labels"].shape == inputs["decoder_input_ids"].shape
+            "labels" in generation_inputs
+            and "decoder_input_ids" in generation_inputs
+            and generation_inputs["labels"].shape == generation_inputs["decoder_input_ids"].shape
         ):
-            inputs = {k: v for k, v in inputs.items() if k != "decoder_input_ids"}
-        generated_tokens = self.model.generate(**inputs, **gen_kwargs)
+            generation_inputs = {k: v for k, v in inputs.items() if k != "decoder_input_ids"}
+        generated_tokens = self.model.generate(**generation_inputs, **gen_kwargs)
```

Comment:
The failed test started failing since `d979cf6e` (months ago 😅 ).
The modification of `inputs` here will be used in the next block related to training.
Since `labels` is in `inputs`, and we moved out `decoder_input_ids`, it tries to create `decoder_input_ids` from `labels` again in `EncoderDecoder` but failed as there is no padding token.

This PR just use `generation_inputs` (without `decoder_input_ids`) for generation but keep `inputs` untouched for training.